segmentstore/server/src/test/java/io/pravega/segmentstore/server/DataRecoveryTestUtils.java

```diff
@@ -0,0 +1,171 @@
+        }
+
+        SegmentToContainerMapper segToConMapper = new SegmentToContainerMapper(debugStreamSegmentContainers.size());
+
+        Iterator<SegmentProperties> it = storage.listSegments();
+        if (it == null) {
+            log.info("No segments found in the long term storage.");
+            return;
+        }
+
+        // Iterate through all segments. Create each one of their using their respective debugSegmentContainer instance.
+        ArrayList<CompletableFuture<Void>> futures = new ArrayList<>();
+        while (it.hasNext()) {
+            SegmentProperties curr = it.next();
+            int containerId = segToConMapper.getContainerId(curr.getName());
+            log.info("Segment to be recovered = {}", curr.getName());
+            metadataSegmentsByContainer.get(debugStreamSegmentContainers.get(containerId)).remove(curr.getName());
+            futures.add(CompletableFuture.runAsync(new SegmentRecovery(debugStreamSegmentContainers.get(containerId), curr)));
+        }
+        Futures.allOf(futures).join();
+
+        for (Map.Entry<DebugStreamSegmentContainer, Set<String>> metadataSegmentsSetEntry : metadataSegmentsByContainer.entrySet()) {
+            for (String segmentName : metadataSegmentsSetEntry.getValue()) {
+                log.info("Deleting segment '{}' as it is not in storage", segmentName);
+                metadataSegmentsSetEntry.getKey().deleteStreamSegment(segmentName, TIMEOUT).join();
+            }
+        }
+    }
+
+    /**
+     * Creates the given segment with the given DebugStreamSegmentContainer instance.
+     */
+    public static class SegmentRecovery implements Runnable {
+        private final DebugStreamSegmentContainer container;
+        private final SegmentProperties storageSegment;
+
+        public SegmentRecovery(DebugStreamSegmentContainer container, SegmentProperties segment) {
+            Preconditions.checkNotNull(container);
+            Preconditions.checkNotNull(segment);
+            this.container = container;
+            this.storageSegment = segment;
+        }
+
+        @Override
+        public void run() {
+            long segmentLength = storageSegment.getLength();
+            boolean isSealed = storageSegment.isSealed();
+            String segmentName = storageSegment.getName();
+
+            log.info("Recovering segment with name = {}, length = {}, sealed status = {}.", segmentName, segmentLength, isSealed);
+            /*
+                1. segment exists in both metadata and storage, re-create it
+                2. segment only in metadata, delete
+                3. segment only in storage, re-create it
+             */
+            val streamSegmentInfo = container.getStreamSegmentInfo(storageSegment.getName(), TIMEOUT)
+                    .thenAccept(e -> {
+                        if (segmentLength != e.getLength() || isSealed != e.isSealed()) {
+                            container.deleteStreamSegment(segmentName, TIMEOUT).join();
```

Comment:
To elaborate -
We want to just update the metadata and not touch the actual segment on LTS. (Except for metadata segment, all other segments should be left alone.)
I fear that calling container.deleteStreamSegment like this will actually trigger request to delete the segment from LTS (it will enqueue operation to delete in BK ledger). Once the new recovered instance restarts then it will reprocess all the data and may end up deleting the data from LTS. 

What we need here is ability to just delete the metadata about the segment and re-create or update it.

Reply:
Using `deleteSegment` from `MetadataStore` now. From the javadoc, this method only deletes a Segment and any associated information from the Metadata Store.